test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -1169,6 +1169,21 @@ protected void ensureStableCluster(int nodeCount, TimeValue timeValue, boolean l
                 + stateResponse.getState());
         }
         assertThat(clusterHealthResponse.isTimedOut(), is(false));
+        ensureFullyConnectedCluster();
+    }
+
+    /**
+     * Ensures that all nodes in the cluster are connected to each other.
+     *
+     * Some network disruptions may leave nodes that are not the master disconnected from each other.
+     * {@link org.elasticsearch.cluster.NodeConnectionsService} will eventually reconnect but it's
+     * handy to be able to ensure this happens faster
+     */
+    protected void ensureFullyConnectedCluster() {
+        for (String node: internalCluster().getNodeNames()) {
```

Comment:
Can you add a method to the test framework that takes an `InternalCluster` and executes this exact logic? You can invoke it here, and in `NetworkDisruption`.